File: homework01/HomeWork01_CSharp/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace HomeWork01_CSharp
{
    public partial class Form2 : Form
    {
        private List<Studente> studentes;
        private Studente studente1;
        private Studente studente2;
        private Studente studente3;

        public Form2()
        {
            InitializeComponent();
            studente1 = new Studente("Rocco", 1.80);
            studente2 = new Studente("Antonio", 1.60);
            studente3 = new Studente("Giovanni", 1.70);
            studentes = new List<Studente>();
            studentes.Add(studente1);
            studentes.Add(studente2);
            studentes.Add(studente3);
        }

        
        private void richTextBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {

            int numberOfStudents = studentes.Count;
            if (numberOfStudents==0)
            {
                richTextBox1.Text = "There are no students to analyze.";
            } else
            {
                double weightSum = 0.0;
                for (int i = 0; i < numberOfStudents; i++)
                {
                    weightSum += studentes[i].GetHeight();
                }
                double averageHeight = weightSum / numberOfStudents;
                richTextBox1.Text = "The students analyzed are:\n " + studente1.GetName() + "\n " + studente2.GetName() + "\n " + studente3.GetName();
                richTextBox1.AppendText("\nThe average height is: \n" + averageHeight.ToString() + " meters.");
            }
        }

        private void Form2_Load(object sender, EventArgs e)
        {
        }
    }

    public class Studente
    {
        private String name;
        private Double height;

        public Studente(String current_name, Double current_height)
        {
            name = current_name;
            height = current_height;
        }

        public string GetName()
        {
            return name;
        }

        public double GetHeight()
        {
            return height;
        }
    }
}

File: homework01/HomeWork01_VB.net/Form2.vb
Imports HomeWork01_VB.net.Form2
Imports HomeWork01_VB.net.HomeWork01_CSharp.Form2

Public Class Form2

    Private students As List(Of Student)
    Private student1 As Student
    Private student2 As Student
    Private student3 As Student

    Private Sub Form2_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        student1 = New Student("Rocco", 1.8)
        student2 = New Student("Antonio", 1.6)
        student3 = New Student("Giovanni", 1.7)
        students = New List(Of Student)
        students.Add(student1)
        students.Add(student2)
        students.Add(student3)
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim numbersOfStudents = students.Count
        If numbersOfStudents = 0 Then
            RichTextBox1.Text = "There are no students to analyze"
        Else
            Dim weightSum As Double = 0.0
            For Each student As Student In students
                weightSum += student.GetHeight()
            Next
            Dim averageHeight = weightSum / numbersOfStudents
            RichTextBox1.Text = "The students analyzed are: " + vbCrLf + student1.GetName() + vbCrLf + student2.GetName() + vbCrLf + student3.GetName()
            RichTextBox1.AppendText(vbCrLf + "The average height is: " + vbCrLf + averageHeight.ToString() + " meters.")
        End If
    End Sub
    Public Class Student
        Private name As String
        Private height As Double
        Public Function GetName() As String
            Return Me.name
        End Function
        Public Function GetHeight() As Double
            Return Me.height
        End Function
        Public Sub New(ByVal Name As String, ByVal Height As Double)
            Me.name = Name
            Me.height = Height
        End Sub
    End Class

End Class




